### ISSUE TITLE
Sbt assembly and build problems

### ISSUE THREAD

**Issue by entity279** [azakordonets/fabricator#23]
Hi, I'm using fabricator with scala 2.11 latest and sbt-assebmbly.

I'm getting a runtime exception which looks like : 

`Caused by: java.lang.NullPointerException
        at java.io.Reader.<init>(Reader.java:78)
        at java.io.InputStreamReader.<init>(InputStreamReader.java:129)
        at scala.io.BufferedSource.reader(BufferedSource.scala:24)
        at scala.io.BufferedSource.bufferedReader(BufferedSource.scala:25)
        at scala.io.BufferedSource.scala$io$BufferedSource$$charReader$lzycompute(BufferedSource.scala:35)
        at scala.io.BufferedSource.scala$io$BufferedSource$$charReader(BufferedSource.scala:33)
        at scala.io.BufferedSource.scala$io$BufferedSource$$decachedReader(BufferedSource.scala:62)
        at scala.io.BufferedSource.mkString(BufferedSource.scala:91)
        at fabricator.entities.RandomDataKeeper$.fabricator$entities$RandomDataKeeper$$parseFile(RandomDataKeeper.scala:31)
        at fabricator.entities.RandomDataKeeper$$anonfun$parseLanguageDataFiles$1.apply(RandomDataKeeper.scala:27)
        at fabricator.entities.RandomDataKeeper$$anonfun$parseLanguageDataFiles$1.apply(RandomDataKeeper.scala:27)
        at scala.collection.immutable.HashSet$HashSet1.foreach(HashSet.scala:316)
        at scala.collection.immutable.HashSet$HashTrieSet.foreach(HashSet.scala:972)
        at fabricator.entities.RandomDataKeeper$.parseLanguageDataFiles(RandomDataKeeper.scala:27)
        at fabricator.entities.RandomDataKeeper$.getJson(RandomDataKeeper.scala:36)
        at fabricator.UtilityService.valuesJson$lzycompute(UtilityService.scala:13)
        at fabricator.UtilityService.valuesJson(UtilityService.scala:13)
        at fabricator.UtilityService.<init>(UtilityService.scala:17)`

Issue went away after compiling and packaging the lib manually along with adding these by hand to our build files 
`  "com.typesafe.play" %% "play-json" % "2.4.8",
  "org.testng" % "testng" % "6.9.4",
  "org.scalatest" %% "scalatest" % "3.0.0-SNAP4" % "test",
  "com.github.nscala-time" %% "nscala-time" % "1.2.0",
  "com.typesafe.scala-logging" %% "scala-logging-slf4j" % "2.1.2",
  "org.slf4j" % "slf4j-api" % "1.7.1",
  "org.reflections" % "reflections" % "0.9.10",
  "ch.qos.logback" % "logback-classic" % "1.0.3",
  "org.assertj" % "assertj-core" % "3.1.0",
  "com.spatial4j" % "spatial4j" % "0.4.1",
  "org.iban4j" % "iban4j" % "2.1.1",
  "commons-validator" % "commons-validator" % "1.4.0",
  "com.google.inject" % "guice" % "3.0",
  "com.github.tototoshi" %% "scala-csv" % "1.1.2",
  "com.google.zxing" % "core" % "3.2.0",
  "io.github.lukehutch" % "fast-classpath-scanner" % "2.0.17"`

**Comment by azakordonets** [comment]
Have you tried to include into you build the latest version ? I did some changes recently. What version of the library did you use ? 

```
resolvers += "Fabricator" at "http://dl.bintray.com/biercoff/Fabricator"

libraryDependencies += "com.github.azakordonets" % "fabricator_2.11" % "2.1.4"
```

**Comment by entity279** [comment]
A colegue has attempted to use 2.1.4 without success apparently.

But yes you're right, I couldn't reproduce the problem anymore after I personally attempted the upgrade. 

However, even w/ fabricator 2.1.4 i'm getting 'Scala Version Problem's errors in  Scala IDE. Regarding the usage of scala-csv-2.10-1.1.1 (while a 2.11 version was expected).

**Comment by azakordonets** [comment]
So, you're still not able to use the library with Scala 2.11 or it's just IDEA complaints about the problem ?

**Comment by entity279** [comment]
I'm able to use the library but the eclipse (not intelliJ IDEA) IDE complains about the scala-csv dependentcy.

We also tested manually inserting the fabricator lib into our project, and copying fabricator's dependencies into our own build.sbt (so that scala-csv is also compiled w/ scala 2-11). That also worked, and also removed the IDE flags. (ofc it's not a nice solution, just saying what we did to get rid of the IDE complains)

**Comment by azakordonets** [comment]
Ok, i'll see if i can update dependencies for scala-csv lib. I will keep you updated

**Comment by azakordonets** [comment]
I'm going to add scala 2.12 support soon and therefore i will probably update libraries for that, maybe it will solve your problems

**Comment by entity279** [comment]
I've stopped using fabricator exactly since I plan on switching to scala 2.12 soon. A static list of names, cities , countries was all I needed anyway

**Comment by azakordonets** [comment]
@entity279 i have just published new version of the library for scala 2.12. If you still need it - feel free to try it out :)
